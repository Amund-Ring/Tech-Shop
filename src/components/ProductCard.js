import React from "react";
import { navigate } from "gatsby";
import * as styles from "../styles/ProductCard.module.css";

function ProductCard({ item }) {
  const handleClick = () => {
    navigate(`/product?id=${item.id}`);
  };

  return (
    <div
      className={styles.productCard}
      onClick={handleClick}
      onKeyDown={handleClick}
      role="button"
      tabIndex={[]}
    >
      <img src={`/products/${item.id}.jpg`} alt={item.name} />
      <div>
        <h2>{`${item.name}`}</h2>
        <h3>{`${item.price},-`}</h3>
      </div>
    </div>
  );
}

export default ProductCard;
