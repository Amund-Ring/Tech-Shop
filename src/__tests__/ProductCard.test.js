import React from "react";
import renderer from "react-test-renderer";
import ProductCard from "../components/ProductCard";

describe("ProductCard", () => {
  it("renders correctly", () => {
    const item = {
      id: 5,
      name: "Bluetooth speaker",
      brand: "JBL",
      price: "800",
      available: false,
      weight: 0.5,
      options: [
        {
          color: ["black"],
          quantity: 15,
        },
        {
          color: ["white"],
          quantity: 0,
        },
        {
          color: ["red"],
          quantity: 1,
        },
      ],
    };

    const tree = renderer.create(<ProductCard item={item} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
