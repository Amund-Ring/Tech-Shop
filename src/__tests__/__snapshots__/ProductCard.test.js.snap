// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard renders correctly 1`] = `
<div
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={Array []}
>
  <img
    alt="Bluetooth speaker"
    src="/products/5.jpg"
  />
  <div>
    <h2>
      Bluetooth speaker
    </h2>
    <h3>
      800,-
    </h3>
  </div>
</div>
`;
